### PR TITLE
feat(yargs): typing to support command module array

### PR DESCRIPTION
**Pull request by kgajera**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/yargs/yargs/blob/main/docs/advanced.md#example-command-hierarchy-using-indexmjs
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/yargs/index.d.ts [hunk 1]
```diff
@@ -176,6 +176,7 @@ declare namespace yargs {
         ): Argv<T>;
         command(command: string | ReadonlyArray<string>, showInHelp: false, module: CommandModule<T, any>): Argv<T>;
         command(module: CommandModule<T, any>): Argv<T>;
+        command(modules: Array<CommandModule<any, any>>): Argv<T>;
```
Comment (forivall) [inline types/yargs/index.d.ts:179]: Let me know if the first type was `any` for a specific reason.

```suggestion
        command(modules: Array<CommandModule<T, any>>): Argv<T>;
```

**Review by forivall — CHANGES_REQUESTED** [review]
just the one question

File: types/yargs/index.d.ts (hunk 1 above)
Comment (kgajera) [reply types/yargs/index.d.ts]: Thanks for the comment and I have made this change.

If you look at this commit, https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64778/commits/2be2b8b845a049672597bd3376a88fd42deb83fc, I did have to remove the generic types from the test in order for it work. It seems like there's some issue with the `command` function overloading that's using the wrong type when I have the generic type defined there.

**Review by forivall — APPROVED** [review]
LGTM! thanks!

**Pull request merged**